### ISSUE TITLE
Group object missing objectId property

### ISSUE THREAD

**Issue by Kayn** [microsoftgraph/msgraph-typescript-typings#7]
When I called the https://graph.windows.net/myorganization/groups?api-version to get all groups, there was a objectId property in the response, but I didn't see it in MicrosoftGraph.Group definition. instead there is a id property which is from Entity interface.

**Comment by dan-silver** [comment]
I believe that's a difference between the [Microsoft Graph](https://graph.microsoft.io/en-us/) and the [Azure AD Graph](https://msdn.microsoft.com/en-us/Library/Azure/Ad/Graph/api/api-catalog).

If I use the AAD graph explorer at https://graphexplorer.cloudapp.net/Home/Index/DemoCompany?UrlRequest=GET&text=https%3A%2F%2Fgraph.windows.net%2Fgraphdir1.onmicrosoft.com%2Fgroups, I do see groups have an objectId field. In the Microsoft Graph, all entities have an `id` property to be consistent between users, groups, files, etc. You can check this on the [Microsoft Graph explorer](https://graph.microsoft.io/en-us/graph-explorer#).

Here's a [blog post](https://blogs.msdn.microsoft.com/aadgraphteam/2016/07/08/microsoft-graph-or-azure-ad-graph/) that offers guidance on when to use Azure AD Graph or the Microsoft Graph.

**Comment by dan-silver** [comment]
@Kayn I'm closing this issue for now, but please reopen if my previous comment didn't solve your issue. Thanks!

**Comment by Kayn** [comment]
thanks for explanation, do you know where i can the typescript types for
Azure AD Graph?

Sincerely
kayn

**Comment by dkershaw10** [comment]
The only fully supported client library for Azure AD Graph is the .Net library.  I would *Strongly* advise you switch to Microsoft Graph (for the reasons provided in the blog post).  Groups is fully supported in MS Graph (and in fact has more functionality than in AAD Graph).  ObjectId was renamed to Id in Microsoft Graph, but semantically (and in underlying data) they are exactly the same.
We are investing heavily in a range of different platforms and languages for our client libraries for Microsoft Graph.  There is no investment in AAD Graph client libraries, and over time very little investment in AAD Graph API as per the blog post.

Let me know if you have more questions.  I'd also like to understand if for some reason you are wedded to using Azure AD Graph, and why.

Hope this helps

**Comment by Kayn** [comment]
Hi Dan,

thanks for your suggestion. we are using the azure AD to authenticate and
authorize our users. and we want to integrate the management of roles,
groups, users into our system. when i did the search, first result came up
is the azure AD graph API, that is why i chose the azure AD graph API.

but if the Microsoft Graph API can do the same thing, I would love to
switch to it.

Thanks again for your help.

Sincerely
kayn

**Comment by dkershaw10** [comment]
Thanks for the response.  I'm also working on making sure that developers who search for materials on accessing directory features land at Microsoft Graph rather than Azure AD Graph, so I'd love to get more details from you on your search experiences.  Can you remember how you searched and landed on Azure AD Graph? 
All Azure AD Graph documentation also has the following banner.  Did you see it?  If so, what changes in the language below would have made you follow the link to Microsoft Graph?

Important
Azure AD Graph API functionality is also available through Microsoft Graph, a unified API that also includes APIs from other Microsoft services like Outlook, OneDrive, OneNote, Planner, and Office Graph, all accessed through a single endpoint with a single access token.

Cheers
Dan.

**Comment by Kayn** [comment]
Hi Dan,

I remembered i searched azure AD API, and checked the first two links.

I saw the notes, but it does not imply any disadvantage to choose azure AD
graph API which has all functionalities i need. and i am not so interested
in the outlook, onedrive......, so i made the decision without any
hesitation. the banner came to my head again until i got your email.

I feels like the words in the banner are not strong enough to change my
mind. more like some thing "good to know".

Sincerely
kayn
